### DISCUSSION TITLE
.csv downloads and datetime/dates

### DISCUSSION THREAD

**Discussion by billcla** [anvilistas/tabulator#143] [Q&A] [answered]
Me again, nearly finished with my project, loving tabulator!

I'm having issues with csv downloads though as datetime and date cells that output to csv in the following format:
`{"v":{"$year":2023,"$month":10,"$day":17,"$hour":9,"$min":3,"$sec":46,"$micro":159000,"$tzinfo":{"$d":{"size":1,"entries":{"!_offset":[{"$mangled":"_offset","$savedKeyHash":"!_offset","v":"_offset"},{"$days":0,"$secs":0,"$micro":0,"$hashcode":-1}]},"buckets":{},"in$repr":false,"$version":1},"$s":[]},"$fold":0,"$hashcode":-1},"$isPyWrapped":true}`

But if I run the csv download option in Anvil it produces what is essentially the string format of the datetime or date as it is in the cell on a repeating table. Am I missing an encoding option? Would prefer to utilise tabulator as its already using the filters and dataset in the client.

example ap below:
https://anvil.works/build#clone:UFTS5UFK75DTAEW4=PTP3SHNJ5FYRWHE25F5EMYJO|null

Would appreciate any hints please! Many thanks.

**Comment by s-cork** [comment, marked as answer]
Yeah - you'll need to use the `Accessor` Module

```python
Tabulator.modules.add("Accessor")
```

Then add this to your date/datetime columns
```python
def accessor_date(value, data, type, params, column, row):
    return value.isoformat()

```

and then in your columns
```python
    {"title": "Today", "field": "some date and time", "headerSort": True, "formatter": "datetime", "formatter_params": 
    {"format": "%d %b %Y, %H:%M:%S"}, "width": 180, "accessor_download": accessor_date},
```

**Reply by billcla** [reply]
That worked perfectly Stu, many thanks! I hadn't understood the use or importance of the accessor module.

**Reply by s-cork** [reply]
Yeah me neither - until recently.
